flask_security/utils.py

```diff
@@ -145,12 +145,16 @@ def verify_and_update_password(password, user):
     :param password: A plaintext password to verify
     :param user: The user to verify against
     """
+    if use_double_hash():
+        verified = (
+            _pwd_context.verify(get_hmac(password), user.password) or
+            _pwd_context.verify(password, user.password)
```

Comment:
@christopherdeutsch there was a problem when the original password was in plaintext.